pkg/scheduler/core/spreadconstraint/group_clusters.go

```diff
@@ -130,7 +133,42 @@ func (info *GroupClustersInfo) generateClustersInfo(clustersScore framework.Clus
 	sortClusters(info.Clusters)
 }
 
-func (info *GroupClustersInfo) generateZoneInfo(spreadConstraints []policyv1alpha1.SpreadConstraint) {
+func (info *GroupClustersInfo) calcGroupScore(clusters []ClusterDetailInfo, rbSpec *workv1alpha2.ResourceBindingSpec) int64 {
+	// First, calculate the base score.
+	var highScoreSum int64
+	var selectedNum int
+	for _, cluster := range clusters {
+		if cluster.Score > info.averageScore {
+			highScoreSum += cluster.Score
+			selectedNum++
+		}
+	}
+	var baseScore int64
+	if selectedNum == 0 {
+		baseScore = clusters[0].Score
+	} else {
+		baseScore = highScoreSum / int64(selectedNum)
+	}
+
+	// Second, calculate the bonus score.
+	var bonusScore int64
+	if rbSpec.Placement == nil ||
```

Comment:
Why does the calculation of Group's score still depend on the scheduling type?

Reply:
Avaiable replicas of every cluster still affects the score of a group, especially `Duplicated` strategy.
eg: The group `G1` has the highest score but low avaiable replicas so that no cluster can meet `rbSepc.replicas` when using `Duplicated` strategy. Should `G1` still keep the highest score?